PythonVisualizations/Array.py

```diff
@@ -258,9 +258,54 @@ def remove(self, val):
             return True
         return False
 
+    def traverse(self):
+        self.cleanUp()
+
+        # draw an index pointing to the first cell
+        indexDisplay = self.createIndex(0, 'j')
+        self.cleanup |= set(indexDisplay)
+
+        # draw output box
+        canvasDimensions = self.widgetDimensions(self.canvas)
+        outputBox = self.canvas.create_rectangle(\
+            ARRAY_X0+ CELL_SIZE//2, ARRAY_Y0 + canvasDimensions[1] // 3, ARRAY_X0 + CELL_SIZE*len(self.list), \
+            ARRAY_Y0 + canvasDimensions[1] // 3 + 50,\
+            fill = OPERATIONS_BG)
+        self.cleanup.add(outputBox)
+
+        for j in range(len(self.list)):
+            # calculate where the value will need to move to
+            outputBoxCoords = self.canvas.coords(outputBox)
+            midOutputBox = (outputBoxCoords[3]-outputBoxCoords[1])//2+outputBoxCoords[1]

```

Comment:
The middle is the average of the two values so this can be simplified as:
`    midOutputBox = (outputBoxCoords[3] + outputBoxCoords[1]) // 2`